src/sentry/api/endpoints/auth_index.py

```diff
@@ -100,8 +112,21 @@ def put(self, request: Request):
             return Response(status=status.HTTP_401_UNAUTHORIZED)
 
         validator = AuthVerifyValidator(data=request.data)
-        if not validator.is_valid():
-            return self.respond(validator.errors, status=status.HTTP_400_BAD_REQUEST)
+        has_valid_sso_session = has_completed_sso(request, request.superuser.org_id)
+        from_superuser_modal = request.data.get("isSuperuserModal")
+
+        if not validator.is_valid() and not has_valid_sso_session:
+            try:
+                auth_identity = AuthIdentity.objects.get(user=request.user)
+                org_id = (
+                    Superuser.org_id
+                    if request.user.is_superuser
```

Comment:
what if a superuser is just trying to use `sudo` and not `superuser`? should we make it so that they may `sudo` without becoming `superuser`?

Reply:
this shouldn't affect if they're trying to sudo or superuser, this just redirects them to their org's sso if they don't have a valid sso session

Reply:
what if they're not in their superuser org? 🙃  i think we've found more test cases to add lol